File: src/domain/Reservation.cs
using System;

namespace AgentiiDeTurism.src.domain
{
    public class Reservation : IIdentifiable<Tuple<string, Trip>>
    {
        public string Client { get; set;  }
        public Trip Trip { get; set; }
        public string PhoneNumber { get; set; }
        public int Seats { get; set; }

        public Reservation(Tuple<string, Trip> id, string phoneNumber, int seats)
        {
            this.Client = id.Item1;
            this.Trip = id.Item2;
            this.PhoneNumber = phoneNumber;
            this.Seats = seats;
        }

        public Tuple<string, Trip> getId()
        {
            return new Tuple<string, Trip>(this.Client, this.Trip);
        }

        public void setId(Tuple<string, Trip> id)
        {
            this.Client = id.Item1;
            this.Trip = id.Item2;
        }

        public override string ToString()
        {
            return "Reservation by " + this.Client + " on trip to " + this.Trip.TouristAttraction + " for " + this.Seats + " seats";
        }
    }
}

File: src/main/java/domain/Reservation.java
package domain;

import utils.Pair;

public class Reservation implements Identifiable<Pair<String, Trip>> {
    private String client;
    private Trip trip;
    private String phoneNumber;
    private int seats;

    public Reservation(Pair<String, Trip> reservationID, String phoneNumber, int seats) {
        this.client = reservationID.getFirst();
        this.trip = reservationID.getSecond();
        this.phoneNumber = phoneNumber;
        this.seats = seats;
    }

    @Override
    public Pair<String, Trip> getId() {
        return new Pair<>(client, trip);
    }

    @Override
    public void setId(Pair<String, Trip> id) {
        client = id.getFirst();
        trip = id.getSecond();
    }

    public String getClient() {
        return client;
    }

    public void setClient(String client) {
        this.client = client;
    }

    public Trip getTrip() {
        return trip;
    }

    public void setTrip(Trip trip) {
        this.trip = trip;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public int getSeats() {
        return seats;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public void setSeats(int seats) {
        this.seats = seats;
    }

    @Override
    public String toString() {
        return "Reservation by " + this.client + " on trip to " + this.trip.getTouristAttraction()
                + " for " + this.seats + " seats";
    }
}
